libraries/standard/mqtt/src/iot_mqtt_api.c

```diff
@@ -469,8 +469,8 @@ static bool _createKeepAliveOperation( const IotMqttNetworkInfo_t * pNetworkInfo
     pMqttConnection->pingreq.u.operation.periodic.ping.nextPeriodMs = ( uint32_t ) keepAliveSeconds * 1000U;
 
     /* Generate a PINGREQ packet. */
-    serializeStatus = _getMqttPingreqSerializer( pMqttConnection->pSerializer )( &( pMqttConnection->pingreq.u.operation.pMqttPacket ),
-                                                                                 &( pMqttConnection->pingreq.u.operation.packetSize ) );
+    serializeStatus = _getMqttPingreqSerializer( pNetworkInfo->pMqttSerializer )( &( pMqttConnection->pingreq.u.operation.pMqttPacket ),
```

Comment:
Wondering why some of these in this file use `pNetworkInfo->pMqttSerializer` while others use `pMqttConnection->pSerializer`

Reply:
This was a bug.  This is during the initialization of the MQTT connection, before the MQTT connection has been set up.  Therefore, the serializer needs to be read from the parameter, pNetworkInfo.